src/main/java/edu/harvard/iq/dataverse/DatasetPage.java

```diff
@@ -2376,10 +2376,11 @@ public String testSelectedFilesForRestrict(){
                 requestContext.execute("PF('selectFilesForRestrict').show()");           
             return "";
         } else {           
-            boolean validSelection = false;
+            boolean validSelection = true;
             for (FileMetadata fmd : selectedFiles) {
-                if (!fmd.isRestricted() ){
-                    validSelection = true;
+                if (fmd.isRestricted() == true) {
+                    validSelection = false;
+                    break;
```

Comment:
These changes just match the logic update in the next method that was done previously - to assure the whole set of files is valid rather than that there is at least one that would be changed.